compiler/pom.xml

```diff
@@ -173,6 +156,41 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>2.3</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <minimizeJar>true</minimizeJar>
+              <artifactSet>
+                <excludes>
+                  <!-- guava which has a consistent API and whose public types we vend in producers -->
+                  <exclude>com.google.guava</exclude>
```

Comment:
Yeah - @tbroyer - we probably need to move away from literals anyway, and I have an internal commit I was toying with to do just that. 

Given that this pull improves the situation of google processors working together, and improves upon no shading, I'm inclined to move forward with this and refine.

I don't think this pull with "cause" problems that aren't already present.  Am I missing something? Right now there's no shading.